CHANGELOG.md

```diff
@@ -42,7 +42,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 
 ### Breaking Changes
 
-- TODO
+- Kafka: Move `sasl` and `tls` config to `TriggerAuthentication` metadata ([#4024](https://github.com/kedacore/keda/pull/4024))
```

Comment:
Is this really a Brekaing Change? I am not sure, @tomkerkhove @JorTurFer wdyt?

Reply:
Not necessarily though. Before we decided to make it backward compatible, I classified it as breaking changes. I can move it out of this section

Reply:
I don't think so, because it's backward compatible

Reply:
Based on the doc changes it looks like a breaking change as we are removing fields/parameters which we already support

Reply:
I'd say that docs are outdated because based on the code, the values are read from both places

Reply:
Yes. values are read from both places. Let me update it. Thanks guys!

Reply:
Thanks. Just our of curiosity, what is the motivation to move this to the Metadata? Because that feels less of data-plane of the CRD to me and just Kubernetes Metadata.

Reply:
It is cumbersome to specify `tls: enable` in base64 in a Secret.

Reply:
So instead of a secret, isn't the real fix that we need a plain text option in a CRD instead of moving this to metadata?

Something like:
```yaml
  plainText:
  - parameter: tls
    value: true   
```

Because it feels better to me to keep everything part of the CRD "body" rather than some in metadata and other in the payload/body.

Reply:
@tomkerkhove Thanks so much for the suggestion. I'm really curious to know how does it look like with your suggestion in `ScaledObject`. :thinking: My guess is below ( pls cmiiw )
```
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: pulsar-scaledobject
  namespace: default
spec:
  scaleTargetRef:
    name: pulsar-consumer
  pollingInterval: 30
  plainText:
  - parameter: tls
    value: true 
```
Hi team,
What do you think about Tom's suggestion?
I'm totally fine with both approaches but I think parking it under metadata would be slightly better in terms of consistency across other scalers ( take [pulsar](https://keda.sh/docs/2.9/scalers/pulsar/) for example )

Reply:
My suggestion is not for ScaledObject, but for Trigger Authentication instead where we already have multiple auth providers and this could be another one

Reply:
Sorry for the delay, I understand what @tomkerkhove proposes here, I have been thinking about the same approach, though I am not 100% sure we should provide a plaintext in TriggerAuth. Users might start using this not ideal approach for storing sensitive data. Or am I too sensitive and we should let them do this? :)

Reply:
@kedacore/keda-core-contributors do you have an opinion on the `plainText` field in TA?